File: 0009-palindrome-number/0009-palindrome-number.ts
function isPalindrome(x: number): boolean {
     // Convert the integer to a string
    const str = x.toString();
    
    // Initialize pointers for the start and end of the string
    let left = 0;
    let right = str.length - 1;
    
    // Continue checking until the pointers meet
    while (left < right) {
        // If characters at left and right pointers are not equal, it's not a palindrome
        if (str[left] !== str[right]) {
            return false;
        }
        // Move the pointers towards the center
        left++;
        right--;
    }
    
    // If the loop completes, the string is a palindrome
    return true;
};
// 12345
// 1221
// 11

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        convert = str(x)
        l = 0
        r = len(convert) - 1

        while (l < r):
            if convert[l] != convert[r]:
                return False
            l += 1
            r -= 1
        return True

        